src/scripts/DataLayerItem.js

```diff
@@ -89,17 +83,22 @@ class Item {
     this._config = itemConfig;
     this._type = (function(config) {
       let type;
-      if (utils.itemConfigMatchesConstraints(config, constraints.dataConfig)) {
-        type = DataLayer.constants.itemType.DATA;
-      } else if (utils.itemConfigMatchesConstraints(config, constraints.eventConfig)) {
+      if (utils.itemConfigMatchesConstraints(config, constraints.eventConfig)) {
         type = DataLayer.constants.itemType.EVENT;
       } else if (utils.itemConfigMatchesConstraints(config, constraints.listenerOnConfig)) {
         type = DataLayer.constants.itemType.LISTENER_ON;
       } else if (utils.itemConfigMatchesConstraints(config, constraints.listenerOffConfig)) {
         type = DataLayer.constants.itemType.LISTENER_OFF;
+      } else if (isPlainObject(config)) {
+        type = DataLayer.constants.itemType.DATA;
       }
       return type;
     }(itemConfig));
+
+    this._config.data = (function(config, type) {
+      return omit(config, Object.keys(constraints[type + 'Config']));
+    })(itemConfig, this._type);
```

Comment:
What's the intention of this logic?
Couldn't it be simplified as follows?
`this._config.data = omit(itemConfig, Object.keys(constraints[this._type + 'Config']));`